snuba/subscriptions/entity_subscription.py

```diff
@@ -0,0 +1,133 @@
+from abc import ABC
+from enum import Enum
+from typing import Any, List, Mapping, Optional, Sequence, Type, Union
+
+from snuba.datasets.entities import EntityKey
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.types import Condition
+
+
+class SubscriptionType(Enum):
+    LEGACY = "legacy"
+    SNQL = "snql"
+    DELEGATE = "delegate"
+
+
+class EntitySubscription(ABC):
+    def __init__(
+        self, subscription_type: SubscriptionType, data_dict: Mapping[str, Any]
+    ) -> None:
+        self.subscription_type = subscription_type
+
+    def get_entity_subscription_conditions(
+        self, offset: Optional[int] = None
+    ) -> Sequence[Union[Expression, Condition]]:
+        return []
+
+    def to_dict(self) -> Mapping[str, Any]:
+        return {}
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, EntitySubscription):
+            return NotImplemented
+        return self.to_dict() == other.to_dict() and isinstance(other, type(self))
```

Comment:
Its necessary in case I want to compare subscriptions .. subscriptions could be practically equivalent but won't be `==` because the entity_subscriptions defined represent different objects

Reply:
You mention `in case`. Are you actually using this today?

Reply:
Yes but only in tests, it is genuinely a pita if the equality function is not there because all tests basically compare some form of subscription objects or subscription objects embedded within task objects, and I have tried to create some test utility function to compare subscription objects but I would need to create one of these for every kind of object that has a subscription in it, which is why I ended up just adding this `__eq__` function to the class.
However, if you have strong opinions against adding this `__eq__` function when it is not required in a non-test environment, I can always just monkey patch the `EntitySubscription` class in tests to have a `__eq__` function

Reply:
Actually scratch that, I couldn't justify adding the `__eq__` just for tests so I just monkey patched the `EntitySubscription` class in tests